pyrokinetics/gk_code/GKOutputReaderTGLF.py

```diff
@@ -0,0 +1,383 @@
+        """
+        Given path to input file, guess at the path for associated output files.
+        For TGLF, simply returns dir of the path.
+        """
+        return Path(filename).parent
+
+    @classmethod
+    def _get_raw_data(
+        cls, dirname: PathLike
+    ) -> Tuple[Dict[str, Any], GKInputTGLF, str]:
+        dirname = Path(dirname)
+        if not dirname.exists():
+            raise RuntimeError(
+                f"GKOutputReaderTGLF: Provided path {dirname} does not exist. "
+                "Please supply the name of a directory containing TGLF output files."
+            )
+        if not dirname.is_dir():
+            raise RuntimeError(
+                f"GKOutputReaderTGLF: Provided path {dirname} is not a directory. "
+                "Please supply the name of a directory containing TGLF output files."
+            )
+
+        # The following list of TGLF files may exist
+        expected_files = {
+            **cls._required_files(dirname),
+            "wavefunction": TGLFFile(dirname / "out.tglf.wavefunction", required=False),
+            "field": TGLFFile(dirname / "out.tglf.field_spectrum", required=False),
+            "ky": TGLFFile(dirname / "out.tglf.ky_spectrum", required=False),
+            "ql_flux": TGLFFile(dirname / "out.tglf.QL_flux_spectrum", required=False),
+            "sum_flux": TGLFFile(
+                dirname / "out.tglf.sum_flux_spectrum", required=False
+            ),
+            "eigenvalues": TGLFFile(
+                dirname / "out.tglf.eigenvalue_spectrum", required=False
+            ),
+        }
+        # Read in files
+        raw_data = {}
+        for key, tglf_file in expected_files.items():
+            if not tglf_file.path.exists():
+                if tglf_file.required:
+                    raise RuntimeError(
+                        f"GKOutputReaderTGLF: The file {tglf_file.path.name} is needed"
+                    )
+                continue
+            # Read in file according to format
+            if key == "ky":
+                raw_data[key] = np.loadtxt(tglf_file.path, skiprows=2)
+
+            else:
+                with open(tglf_file.path, "r") as f:
+                    raw_data[key] = f.read()
+
+        input_str = raw_data["input"]
+        gk_input = GKInputTGLF()
+        gk_input.read_str(input_str)
+        return raw_data, gk_input, input_str
+
+    @classmethod
+    def _init_dataset(
+        cls, raw_data: Dict[str, Any], gk_input: GKInputTGLF
+    ) -> xr.Dataset:
+        """
+        Sets coords and attrs of a Pyrokinetics dataset from a collection of TGLF
+        files.
+
+        Args:
+            raw_data (Dict[str,Any]): Dict containing TGLF output.
+            gk_input (GKInputTGLF): Processed TGLF input file.
+
+        Returns:
+            xr.Dataset: Dataset with coords and attrs set, but not data_vars
+        """
+
+        if gk_input.is_linear():
+            f = raw_data["wavefunction"].split("\n")
```

Comment:
No need to change, but `str.splitlines` is a bit more explicit

Reply:
That is better, makes looking for the eigenvalues in `out.tglf.run` a bit cleaner